nix/scripts/shell.sh

```diff
@@ -16,52 +15,57 @@ export TERM=xterm # fix for colors
 shift # we remove the first -c from arguments
 
 if [[ -z "${TARGET}" ]]; then
-    export TARGET="default"
-    echo -e "${YLW}Missing TARGET, assuming default target.${RST} See nix/README.md for more details." 1>&2
+  export TARGET="default"
+  echo -e "${YLW}Missing TARGET, assuming default target.${RST} See nix/README.md for more details." 1>&2
 fi
 
 # Minimal shell with just Nix sourced, useful for `make nix-gc`.
 if [[ "${TARGET}" == "nix" ]]; then
-    eval $@
-    exit 0
+  eval $@
+  exit 0
 fi
 if [[ -n "${IN_NIX_SHELL}" ]] && [[ -n "${NIX_SHELL_TARGET}" ]]; then
-    if [[ "${NIX_SHELL_TARGET}" == "${TARGET}" ]]; then
-        echo -e "${YLW}Nix shell for TARGET=${TARGET} is already active.${RST}" >&2
-        exit 0
-    else
-        # Nesting nix shells does not work due to how we detect already present shell.
-        echo -e "${RED}Cannot nest Nix shells with different targets!${RST}" >&2
-        exit 1
-    fi
+  if [[ "${NIX_SHELL_TARGET}" == "${TARGET}" ]]; then
+    echo -e "${YLW}Nix shell for TARGET=${TARGET} is already active.${RST}" >&2
+    exit 0
+  else
+    # Nesting nix shells does not work due to how we detect already present shell.
+    echo -e "${RED}Cannot nest Nix shells with different targets!${RST}" >&2
+    exit 1
+  fi
 fi
 
 entryPoint="default.nix"
 nixArgs=(
-    "--show-trace"
-    "--attr shells.${TARGET}"
+  "--show-trace"
+  "--attr shells.${TARGET}"
 )
 
 config=''
 if [[ -n "${STATUS_GO_SRC_OVERRIDE}" ]]; then
-    config+="status-im.status-go.src-override=\"${STATUS_GO_SRC_OVERRIDE}\";"
+  config+="status-im.status-go.src-override=\"${STATUS_GO_SRC_OVERRIDE}\";"
 fi
 config+="status-im.build-type=\"${BUILD_TYPE}\";"
 
 if [[ -n "$config" ]]; then
-    nixArgs+=("--arg config {$config}")
+  nixArgs+=("--arg config {$config}")
 fi
 
 # This variable allows specifying which env vars to keep for Nix pure shell
 # The separator is a colon
 if [[ -n "${_NIX_KEEP}" ]]; then
-    nixArgs+=("--keep ${_NIX_KEEP//,/ --keep }")
+  nixArgs+=("--keep ${_NIX_KEEP//,/ --keep }")
 fi
 
 # Not all builds are ready to be run in a pure environment
 if [[ -n "${_NIX_PURE}" ]]; then
-    nixArgs+=("--pure")
-    pureDesc='pure '
+  nixArgs+=("--pure")
+  pureDesc='pure '
+fi
+
+if [[ "|android-sdk|android|gradle|keytool|status-go|" =~ "|$TARGET|" ]]; then
```

Comment:
Shouldn't the group matching be enclosed in round brackets?
```suggestion
if [[ "(android-sdk|android|gradle|keytool|status-go)" =~ "${TARGET}" ]]; then
```
Also, please use curly braces for variables interpolation.

Reply:
It's not really a group matching. In case of group matching the following will pass: "androi", where in my case only strict full words.

Reply:
Will add curly braces.

Reply:
I think the regex works if it's on the right side, and not quoted:
```
 > function test() { [[ "$1" =~ ^(android-sd|android|gradle|keytool|status-go)$ ]] && echo MATCH }             
 > match androidd                                                                                 

 > match status

 > match android 
MATCH
 > match gradle 
MATCH
```

Reply:
Agree, regex is more intuitive, applied.